backbone.js

```diff
@@ -1464,7 +1563,22 @@
     // Execute a route handler with the provided parameters.  This is an
     // excellent place to do pre-route setup or post-route cleanup.
     execute: function(callback, args, name) {
-      if (callback) callback.apply(this, args);
+      var router = this;
+      var error = function(err) {
+        router.trigger('error', err, name, args, callback);
+      };
+      if (callback) {
+        try {
+          var route = callback.apply(this, args);
+          if (isPromise(route)) {
+            route['catch'](error);
+          }
+        } catch(e) {
+          error(e);
+        }
+      } else {
+        error();
+      }
```

Comment:
Are you talking about jmeas/backbone.base-router? I should like to check it out :)